File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    score = [0,0,0,0]
    forward_category  = ["RT","CF","JM","AN"]
    reverse_category = ["TR","FC","MJ","NA"]
    for i in range(len(survey)):
        if survey[i] in forward_category:
            index = forward_category.index(survey[i])
            score[index] += scoring(choices[i])
        else:
            index = reverse_category.index(survey[i])
            score[index] += - scoring(choices[i])
    answer = ["T","E","S","T"]
    for i in range(4):
        if score[i] >= 0:
            answer[i] = forward_category[i][0]
        else:
            answer[i] = forward_category[i][1]
    answer = "".join(answer)  
    return answer


def scoring(category_score):
    return -(category_score - 4)
    

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        int[] point = new int[8];
        int temp = -3;
        for (int i =1; i < point.length; i++) {
            point[i] = temp;
            temp++;
        }
        
        int[] answer = new int[4];
        for (int i = 0; i<choices.length; i++) {
            String serveyTemp = survey[i];
            int takePoint = point[choices[i]];
            if (serveyTemp.equals("RT")) {
                answer[0] += takePoint;
            } else if (serveyTemp.equals("TR")) {
                answer[0] -= takePoint;
            } else if (serveyTemp.equals("CF")) {
                answer[1] += takePoint;
            } else if (serveyTemp.equals("FC")) {
                answer[1] -= takePoint;
            } else if (serveyTemp.equals("JM")) {
                answer[2] += takePoint;
            } else if (serveyTemp.equals("MJ")) {
                answer[2] -= takePoint;
            } else if (serveyTemp.equals("AN")) {
                answer[3] += takePoint;
            } else if (serveyTemp.equals("NA")) {
                answer[3] -= takePoint;
            } 
        }
        
        StringBuilder answerString = new StringBuilder();
        if (answer[0] > 0) {
            answerString.append("T");
        } else if (answer[0] < 0) {
            answerString.append("R");
        } else {
            answerString.append("R");
        }
        
        if (answer[1] > 0) {
            answerString.append("F");
        } else if (answer[1] < 0) {
            answerString.append("C");
        } else {
            answerString.append("C");
        }
        
        if (answer[2] > 0) {
            answerString.append("M");
        } else if (answer[2] < 0) {
            answerString.append("J");
        } else {
            answerString.append("J");
        }
        
        if (answer[3] > 0) {
            answerString.append("N");
        } else if (answer[3] < 0) {
            answerString.append("A");
        } else {
            answerString.append("A");
        }
        
        return answerString.toString();
    }
}